sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.net6.0.cs

```diff
@@ -723,6 +725,13 @@ public enum BlobGeoReplicationStatus
         Bootstrap = 1,
         Unavailable = 2,
     }
+    public partial class BlobGetPropertiesOptions
+    {
+        public BlobGetPropertiesOptions() { }
+        public Azure.Storage.Blobs.Models.BlobRequestConditions Conditions { get { throw null; } set { } }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public bool? UserPrincipalName { get { throw null; } set { } }
```

Comment:
Should we call this `DataLakeUserPrincipalName` everywhere to make the purpose a little clearer to anyone who might notice?

Reply:
We might need to work on the name of the property too.  I would expect something called `Name` to be a `string`.  What does the `bool` imply here?

Reply:
A good explanation is here - https://github.com/Azure/azure-sdk-for-net/pull/40592/files#diff-d31f158a309e368ae6a80b01d55a48e00fc8f33adf35b4fa25b46cf593b34541R32

This is a boolean the customer can set.  If true, Owner, Group, and Permissions will be returned as User Principal Names instead of Azure Active Directory Object IDs.

I do agree we could work on the name of the parameter.